app-misc/kryoflux-dtc/kryoflux-dtc-2.6.ebuild

```diff
@@ -0,0 +1,115 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MY_P="kryoflux_${PV}_linux"
+
+inherit desktop udev xdg-utils
+
+DESCRIPTION="KryoFlux Host Software"
+HOMEPAGE="https://www.kryoflux.com"
+SRC_URI="https://www.kryoflux.com/download/${MY_P}.tar.bz2
+	java? ( https://www.kryoflux.com/kryoflux-ui.jar )"
+
+KEYWORDS="-* ~amd64 ~x86"
+LICENSE="SPS"
+SLOT="0"
+IUSE="demos doc fast-firmware java static"
+
+RDEPEND="dev-libs/spsdeclib
+	virtual/libusb:1
+	virtual/udev
+	java? ( virtual/jre )"
+
+DEPEND="app-arch/unzip"
+
+RESTRICT="bindist mirror strip"
+
+S="${WORKDIR}/${MY_P}"
+
+QA_PREBUILT="/usr/bin/kryoflux-dtc"
+
+src_unpack() {
+	# Unpack archive
+	unpack ${MY_P}.tar.bz2
+
+	if use java; then
+		# Updated *.jar for compatibility with >= Java 9
+		cp "${DISTDIR}"/kryoflux-ui.jar "${S}" || die
+
+		# Extract kryoflux-ui.jar to get a logo for the meny entry
+		unpack kryoflux-ui.jar
+	fi
+}
+
+src_install() {
+	# Choose ARCH and install binary
+	newbin dtc/$(usex amd64 x86_64 i686)/$(usex static static dynamic)/dtc kryoflux-dtc
+
+	# Install udev rule
+	cat <<-EOF > "${T}"/80-kryoflux.rules || die
+	ACTION=="add", SUBSYSTEM=="usb", ATTR{idVendor}=="03eb", ATTR{idProduct}=="6124", GROUP="floppy", MODE="0660"
+	EOF
+	udev_dorules "${T}"/80-kryoflux.rules
+
+	# Install UI
+	if use java; then
+		# Install Java UI
+		insinto /usr/share/kryoflux-ui
+		doins kryoflux-ui.jar
```

Comment:
Won't kryoflux-ui be confused about the renamed executable? Or is it stand-alone?

Reply:
Can't fully test right now. But I think, this won't work, when I look at the strace output.
Hm, any suggestions? This only affects the *.jar GUI.

Standalone CLI dtc will always work fine with a different name.

`
test.log.1467:stat("./dtc", 0x7fa538efb600)           = -1 ENOENT (No such file or directory)
test.log.1467:stat("/home/<username>/bin/dtc", 0x7fa538efb370) = -1 ENOENT (No such file or directory)
test.log.1467:stat("/home/<username>/.local/bin/dtc", 0x7fa538efb370) = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/local/sbin/dtc", 0x7fa538efb370) = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/local/bin/dtc", 0x7fa538efb370) = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/sbin/dtc", 0x7fa538efb370)   = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/bin/dtc", 0x7fa538efb370)    = -1 ENOENT (No such file or directory)
test.log.1467:stat("/sbin/dtc", 0x7fa538efb370)       = -1 ENOENT (No such file or directory)
test.log.1467:stat("/bin/dtc", 0x7fa538efb370)        = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/games/dtc", 0x7fa538efb370)  = -1 ENOENT (No such file or directory)
test.log.1467:stat("/usr/local/games/dtc", 0x7fa538efb370) = -1 ENOENT (No such file or directory)
test.log.1467:stat("/snap/bin/dtc", 0x7fa538efb370)   = -1 ENOENT (No such file or directory)
`

Reply:
Maybe you could change the file inside and repack the .jar? @gentoo/java might help. Using regular name with a blocker would also work but `sys-apps/dtc` is required by qemu, so a lot of people are going to be affected.